### ISSUE TITLE
Any intrest in bringing this to ff/palemoon

### ISSUE THREAD

**Issue by alidan** [deanoemcke/thegreatsuspender#409]
I'm seeing if I can migrate to Firefox due to chrome becoming more and more unusable due to its updates demanding i rest my computer till it randomly fixes itself. 

There is a lot that firefox/palemoon can do, but there is no working tab suspender that works even half as well as this does. There is a lot I can live without, but this extension is one that I require, or at least the functionality (being able to scroll through tabs without them reloading, or even a keyboard shortcut for suspending a tab)

**Comment by xMajedz** [comment]
i agree and since firefox is copying chrome extension method it should be easier now

**Comment by alidan** [comment]
Funny thing is, because firefox is moving to chrome style, its getting rid of all its advantages, taking on chrome's disadvantages along with not fixing the core of firefox. Kinda kills all use firefox had, at least the forks don't have to abandon the extensions though.

**Comment by xMajedz** [comment]
yeah its really disappointing a lot of extensions i use will be effected with it.
but i'm trying to look to the bright side through this. ( which i don't where it is )

**Comment by andreicristianpetcu** [comment]
I made an unofficial WebExtension [build for Firefox](https://andreicristianpetcu.github.io/chrome2firefox/addons/unofficial_the_great_suspender-6.21-an+fx.xpi). It does not work exactly as in Chrome but it should be decent. I made no change to the extension from Chrome Web Store. I just renamed it and signed it. It basically "Just works". Please try it in [Firefox Nightly](https://www.mozilla.org/en-US/firefox/nightly/all/) or [DevEdition](https://www.mozilla.org/en-US/firefox/developer/) since Firefox stable might have some missing APIs. Enjoy!

**Comment by andreicristianpetcu** [comment]
For anyone wanting to hack on this issue and actually make an official build I think the best place to start is by going to the `about:debugging` page in Firefox after you have installed the addon I mentioned above. Also [webextension-polyfill](https://github.com/mozilla/webextension-polyfill) might be useful.

**Comment by jonathanKingston** [comment]
@deanoemcke this would be great to have support, since this bug was raised it seems the issues in supporting Firefox have become just browser rendering issues and a few edge cases. I did a quick debug and most things appear to work.

**Comment by deanoemcke** [comment]
(from the chrome webstore)

Works In Opera Except For Hotkeys
Love this extension, and works well in Opera. I have several windows and at times over 100 tabs. (I have issues :( ) This avoids the browser from crashing.

The path the hotkeys are saved to seems to be a constant to the chrome folder.

If you make that browser dependent it would be would make it 100% compliant with Opera as far as I know.

You can use the context menus in anyways, so this is just to let you know it works very well in Opera more then anything.

**Comment by diego898** [comment]
Just wanted to check back up on this? I think this is a must have for those us finally trying to migrate to chrome from ff!

**Comment by deanoemcke** [comment]
To chrome from ff? Or the other way around?
Yeah, it is still on my radar. But after the new release..

**Comment by liamjohnston** [comment]
+1 but definitely sometime after the next release. I am one of the many devs looking to make the move to Firefox, and my use of (and involvement in) TGS is one of the main things holding me back.

**Comment by shimizurei** [comment]
What is the status on this feature?

**Comment by guillempuche** [comment]
I'm switching to Firefox due to user privacy.

Will you @deanoemcke develop Great Suspender extension?

**Comment by arnoldsandoval** [comment]
+1 this is probably the _only_ thing I miss about chrome.

**Comment by dvalter** [comment]
For anyone who's still interested in porting it over to Firefox: 
I tried to solve or mitigate most of FF incompitabilities I've seen.
Unfortunately I could not make it work for container tabs (API's broken, I use discard instead) and private/incognito mode (API's broken even worse, it requires significant code rewrite).
Another issue I've ran into is handling more than few hundred suspended tabs on start. I'm not sure why number of tabs make *that* much difference for a startup code.

Code is available [here](https://github.com/dvalter/thegreatsuspender).
Currently it should not work in Chromuim-based browsers, but if @deanoemcke wants, I'll put effort to merge it into upstream.

**Comment by ItzMichaelPhillips** [comment]
@dvalter Dude you're a life saver, tysm
